a-short-introduction-to-kubernetes/index.md

```diff
@@ -0,0 +1,372 @@
+```
+
+## ReplicaSets
+
+An application like Directus works with replicas if you set up it with external file storage, and you are not using SQLLite (there are still ways to have replicas with internal file storage and SQLite, but that is more advanced then I will cover in this post, and I don't recommend you do that).
+
+Here we are creating 3 replicas of Directus:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: directus-Deployment
+  labels:
+    app: directus
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: directus
+  template:
+    metadata:
+      labels:
+        app: directus
+    spec:
+      containers:
+      - name: directus
+        image: directus:1.32.3
+        ports:
+          - containerPort: 80
+```
+
+## Volumes and StorageClasses
+
+As you could see in the MariaDB StatefulSets above, we are mounting volumes. Volumes could be temporarily, like a temp dir, or they could be persistent and data stays between Deployments.
+
+A persistent volume needs a StorageClass, so it could create the volume, and a StorageClass normally has the type `ReadWriteOnce` - which means it only could be written to by one pod (but all of the containers in the pod). Another type of StorageClass could use `ReadWriteMany` - and that is useful if you have many pods that needs to write to the same filesystem - like uploading files. You should never setup a database with a ReadWriteMany StorageClass, that could end badly.
+
+Here is an example of a Deployment with an `emptyDir` (`/tmp`) - a non-persistent volume, when the pod restarts, it is empty again.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: directus-Deployment
+  labels:
+    app: directus
+spec:
+  selector:
+    matchLabels:
+      app: directus
+  template:
+    metadata:
+      labels:
+        app: directus
+    spec:
+      containers:
+      - name: directus
+        image: directus/directus:10.8.3
+        ports:
+          - containerPort: 80
+        env:
+          - name: PUBLIC_URL
+            value: https://directus.app
+          volumeMounts:
+            - name: tmp
+              mountPath: /tmp
+      volumes:
+        - name: tmp
+          emptyDir: {}
+```
+
+## ConfigMaps
+
+ConfigMaps are used to store different kind of config objects, it could be environment variables, or like a whole file, which later could be used to mount (like a volume), and the application uses that.
+
+Like config for MariaDB:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: release-name-mariadb
+  labels:
+    app: mariadb
+data:
+  my.cnf: |-
+    [mysqld]
+    skip-name-resolve
+    max_allowed_packet=16M
+```
+
+And then mounted:
+
+```yaml
+[...]
+          volumeMounts:
+            - name: config
+              mountPath: /opt/bitnami/mariadb/conf/my.cnf
+              subPath: my.cnf
+      volumes:
+        - name: config
+          configMap:
+            name: release-name-mariadb
+[...]
+```
```

Comment:
Fix indentation

Reply:
spaces are her intentional, as this is further in in the file. should I keep them?

Reply:
Oh I see. I have a better way to present these, but i'll handle it when publishing.